api/src/main/java/org/apache/iceberg/Snapshot.java

```diff
@@ -68,23 +69,58 @@ public interface Snapshot extends Serializable {
    * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
    *
    * @return a list of ManifestFile
+   * @deprecated since 1.0.0 - Use {@link Snapshot#allManifests(FileIO)} instead.
    */
+  @Deprecated
   List<ManifestFile> allManifests();
 
+  /**
+   * Return all {@link ManifestFile} instances for either data or delete manifests in this snapshot.
+   * <p>
+   * The FileIO instance is passed in for cases where the correct FileIO to use is not known when the
+   * TableMetadata (and thus BaseSnapshot implementations) is not known, such as parsing JSON for the
+   * RESTCatalog or for highly sessionzed applications like Trino or shared Spark servers.
```

Comment:
I don't think this paragraph is needed. The FileIO is passed in because it doesn't make sense for `Snapshot` to have one. The reason why we didn't require it before is because we originally kept manifests in the Snapshot metadata itself.

Reply:
Removed.